### DISCUSSION TITLE
Continuity in secondary domain from mesh file input

### DISCUSSION THREAD

**Discussion by AnthonyB08** [idaholab/moose#17991] [Q&A Meshing] [answered]
Hi all,

I am trying to import a mesh file of a pipe with some thickness. This then becomes a two-domain problem. Block id, 3, is the inner volume of the pipe, while block Id 4 is the thickness of the pipe. The interface between the two domains is side set 13 and   12. I would like to impose continuity between the domains, which will require merging these side sets to form an interface or implement a continuity boundary condition at these side sets.

Side set :13 Block Id =3 
![image](https://user-images.githubusercontent.com/55299169/120563800-76982700-c3d7-11eb-8dbd-3d9795bea561.png)

Side set :12 Block Id =4
![image](https://user-images.githubusercontent.com/55299169/120563853-93ccf580-c3d7-11eb-9fd1-117ed2915bda.png)

**Comment by AnthonyB08** [comment, marked as answer]
P.S The inner domain was hidden for best illustration. See the mesh of the Side Set depicted as orange.

**Reply by jiangwen84** [reply]
Potentially, you can use glued contact to enforce the continuity of two blocks. However, it would be much easier to create conforming mesh for two blocks, any reason that you can not do it using your meshing tools?

**Reply by AnthonyB08** [reply]
It is quite a lengthy progress until I can get meshio , from python, to work. I design and mesh my domains in Ansys Fluent. Then import it into Coreform to export as a .e (exodus file). I have glued the two interfaces together but it is then only reading a single block (domain).  

![image](https://user-images.githubusercontent.com/55299169/120686229-c4ac3980-c46e-11eb-8701-f17d28f82c4d.png)


I will have to read around.

**Comment by GiudGiud** [comment]
Hi Anthony

You could use a Parsed meshgenerator to separate the two blocks in moose
https://mooseframework.inl.gov/source/meshgenerators/ParsedSubdomainMeshGenerator.html

Guillaume
